### ISSUE TITLE
Boxed types

### ISSUE THREAD

**Issue by alexbirkett** [agensdev/playbook-android#8]
When to use, when not to use and how to avoid unnecessary use.

**Comment by Ehyeh-Asher-Ehyeh** [comment]
I can really only think of one case for when to use: when parsing json, or similar where the primitive-analogues are nullable, to a java object representation.

**Comment by hfossli** [comment]
In kotlin you'll be able to use nullable values, right? So it could basically replace boxed types for json in that manner, no?

**Comment by Ehyeh-Asher-Ehyeh** [comment]
Yes, but the end result is the same: since kotlin compiles to java byte code, nullable primitives in kotlin are represented as boxed primitives in the compiled code.
